### ISSUE TITLE
Create basic Photo Handler

### ISSUE THREAD

**Issue by Tjzabel** [RITlug/teleirc#195]
We need a way to share images sent on Telegram to IRC.

**Comment by Tjzabel** [comment]
Currently blocked on #177 until we can determine which Imgur library to use.

**Comment by jwflory** [comment]
Note #41: since we will likely write our own Imgur handler, we should do some sort of cache lookback if a photo message is edited. We shouldn't upload an image from an edited message twice; we should reuse the uploaded image if a user edits the photo caption.

**Comment by Tjzabel** [comment]
@jwflory as a reply to your comment, that will not require a cache lookback. 

We could check to see if the TG Photo message contains an edit. If so, send a message with the updated caption. Something like:
```
jwflory updated the photo caption: "My updated caption"
```

**Comment by Tjzabel** [comment]
Also, I think we can start work on implementing the basics of this issue similar to the document handler. Then, issue #177 can focus on making use of whichever hosting platform we choose.

**Comment by jwflory** [comment]
> We could check to see if the TG Photo message contains an edit. If so, send a message with the updated caption. Something like:
> 
> ```
> jwflory updated the photo caption: "My updated caption"
> ```

My only concern is if I sent a group of 3-4 pictures with different captions, and I edit one of them, it might not be clear which photo it is. Also, two photos could be sent in a chat, one from yesterday and one from ten minutes ago. If I edit the older one, it might appear that I edited the caption of the newer one when in fact I did not.

> Also, I think we can start work on implementing the basics of this issue similar to the document handler. Then, issue #177 can focus on making use of whichever hosting platform we choose.

Sounds good in theory to me, so long as the handler implementation is agnostic to whatever platform we choose, since we were talking about also writing our own Imgur image library.

**Comment by Tjzabel** [comment]
@jwflory I put work into the base Telegram photo handler. It recognizes when a photo has been sent, and if IRC allows it, sends over the photo caption if there is one.

I think the issue with editing photo and document captions is a separate issue, since message edits encapsulate a lot more than just photos.

**Comment by jwflory** [comment]
Thanks for the update @Tjzabel. Should we assign this issue to you then?

> and if IRC allows it

Could you clarify what you mean if IRC allows it?

> since message edits encapsulate a lot more than just photos

You're not wrong. However I think this is still specific to the photo handler though, since we are essentially tying three pieces of data together: a Telegram message ID, an Imgur URL, and a photo caption.

However I think the way to go here is to split this into a v2.0.0 and v2.x.x issue. For v2.0.0, we should aim for feature parity with v1.x.x, which was uploading the image every time. For a v2.x.x release, we could look at using something like an in-memory database of SQLite for something like this as a basic form of caching.

If this makes sense to you, I'll reopen #41 and we can track that for a v2.x.x release.

**Comment by Tjzabel** [comment]
@jwflory "If IRC allows it" meaning if we have a config option to allow sending photos to IRC, but iirc we removed that config option.

I suppose you can open #41 back up. An easy fix would be just sending the date alongside the updated caption, so users can at least match the date/time with the correct image. But, we can debate specific implementations within that issue.
```
jwflory updated the photo caption sent on Jan 1, 2020 at 8pm: "My updated caption"
```
